kubernetes/config/kube_config_util.c

```diff
@@ -39,21 +44,39 @@ char *kubeconfig_mk_cert_key_tempfile(const char *data)
         cert_key_data_bytes = strlen(cert_key_data);
     }
 
+#ifndef _WIN32
     char tempfile_name_template[] = KUBE_CONFIG_TEMPFILE_NAME_TEMPLATE;
     int fd = mkstemp(tempfile_name_template);
+#else
+    char *tempdir_env = getenv("TEMP");
+    int tmpFileNameSize = strlen(tempdir_env) + strlen(KUBE_CONFIG_TEMPFILE_NAME_TEMPLATE) + 1;
+    char *tempfile_name_template = calloc(tmpFileNameSize, sizeof(char));
+    snprintf(tempfile_name_template, tmpFileNameSize, KUBE_CONFIG_TEMPFILE_NAME_TEMPLATE, tempdir_env);
+    char *filename = _mktemp(tempfile_name_template);
+    int fd = _sopen( filename, O_CREAT|O_WRONLY, _SH_DENYWR );
+#endif
+
     if (-1 == fd) {
         fprintf(stderr, "%s: Creating temp file for kubeconfig failed with error [%s]\n", fname, strerror(errno));
+        free(tempfile_name_template);
```

Comment:
cannot free tempfile_name_template on Linux

Reply:
Now it should be fixed